### DISCUSSION TITLE
All outgoing requests are failing post Chaos experiment

### DISCUSSION THREAD

**Discussion by malaykeshav** [chaos-mesh/chaos-mesh#3685] [Q&A]
We recently ran a few pod-networkchaos and pod-httpchaos (packet loss, delay, request abort, request delay, response delay, response abort) on our service deployment. But post the experiment, we are seeing that all outgoing requests from the pods (1 service) we were testing are now failing. I believe chaos-mesh did not gracefully undo some settings and now the service is broken. During the experiment we saw a lot of :

```
2022-10-05T01:08:04.789Z	INFO	records	recover chaos	{"id": "development/central-556d7b45bf-h7d7c"}
2022-10-05T01:08:04.789Z	ERROR	records	fail to recover chaos	{"error": "rpc error: code = DeadlineExceeded desc = context deadline exceeded"}
github.com/chaos-mesh/chaos-mesh/controllers/common/pipeline.(*Pipeline).Reconcile
	/mnt/controllers/common/pipeline/pipeline.go:73
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).reconcileHandler
	/tmp/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.9.5/pkg/internal/controller/controller.go:298
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem
	/tmp/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.9.5/pkg/internal/controller/controller.go:253
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func2.2
	/tmp/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.9.5/pkg/internal/controller/controller.go:214
```

and 
```
2022-10-05T01:02:59.010Z	ERROR	records	fail to apply chaos	{"error": "rpc error: code = DeadlineExceeded desc = context deadline exceeded"}
github.com/chaos-mesh/chaos-mesh/controllers/common/pipeline.(*Pipeline).Reconcile
	/mnt/controllers/common/pipeline/pipeline.go:73
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).reconcileHandler
	/tmp/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.9.5/pkg/internal/controller/controller.go:298
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem
	/tmp/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.9.5/pkg/internal/controller/controller.go:253
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func2.2
	/tmp/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.9.5/pkg/internal/controller/controller.go:214
```

What can be done to recover from this? None of the experiments are running anymore. The dashboard shows empty. We have tried deleting all experiments and putting them in the archive, but it has not helped.

--------------------
EDIT:

Redeploying with a newer docker image of the service fixed the issue. (I wonder if the iptable changes are related to the docker image?)
We would still like to understand why this happened and how we can fix this if it happens in the future as it had heavy consequences with service being unavailable.

**Comment by STRRL** [comment]
Please do not use `Pod<Something>Chaos` directly,  they are internal CRDs. Use `NetworkChaos` and `HTTPChaos` with certain `selector` field as instead.

**Reply by malaykeshav** [reply]
> Please do not use `Pod<Something>Chaos` directly, they are internal CRDs. Use `NetworkChaos` and `HTTPChaos` with certain `selector` field as instead.

Yes, We used NetworkChaos and HTTPChaos. It showed up as `pod-networkchaos` and `pod-httpchaos` in logs, hence I mentioned. Rest assured we used NetworkChaos and HTTPChaos.
